plugins/modules/net_tools/dnsimple.py

```diff
@@ -201,113 +200,134 @@ def main():
     priority = module.params.get('priority')
     state = module.params.get('state')
     is_solo = module.params.get('solo')
+    pagination_per_page = 30
 
-    if account_email and account_api_token:
-        client = DNSimple(email=account_email, api_token=account_api_token)
-    elif os.environ.get('DNSIMPLE_EMAIL') and os.environ.get('DNSIMPLE_API_TOKEN'):
-        client = DNSimple(email=os.environ.get('DNSIMPLE_EMAIL'), api_token=os.environ.get('DNSIMPLE_API_TOKEN'))
-    else:
-        client = DNSimple()
+    try:
+        if account_email and account_api_token:
+            client = Client(email=account_email, access_token=account_api_token)
+        elif os.environ.get('DNSIMPLE_EMAIL') and os.environ.get('DNSIMPLE_API_TOKEN'):
+            client = Client(email=os.environ.get('DNSIMPLE_EMAIL'), access_token=os.environ.get('DNSIMPLE_API_TOKEN'))
+        else:
+            client = Client()
+        account = client.identity.whoami().data.account
+
+        # user supplied a user token instead of account api token
+        if not account:
+            accounts = Accounts(client).list_accounts().data
+            if len(accounts) != 1:
+                module.fail_json(msg="Provided dnsimple token is a user token with multiple accounts. Use an account token or a user token with access to a single account. See https://support.dnsimple.com/articles/api-access-token/")
```

Comment:
hi @rekup you need to break it down in smaller chunks, this line is too long and that is tripping the sanity checks